apps/src/blockly/customBlocks/googleBlockly/mutators/procedureDefMutator.ts

```diff
@@ -133,19 +127,22 @@ export const procedureDefMutator = {
    * Accepts a JSON serializable state value and applies it to the block.
    * @param state The state to apply to this block (see saveExtraState above).
    */
-  loadExtraState: function (state) {
+  // TODO: define a better type for state.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  loadExtraState: function (this: ProcedureBlock, state: Record<string, any>) {
```

Comment:
Can we not just refer to state's type as an object?

Reply:
`object` is an empty object, so typescript will complain when we try to access things like `state['procedureId']`

Reply:
I see. Would we then want to (eventually) define an interface that list out all the potential properties (`procedureId`, `params`, etc.) with their types?

Reply:
yep that would be ideal, I figured that was a reasonable thing to do as a follow up.

Reply:
I agree